Protocol.md

```diff
@@ -106,7 +106,7 @@ When using a transport protocol that does not provide compatible framing, the Fr
     +-----------------------------------------------+
 ```
 
-* __Frame Length__: (24 = max 16,777,216 bytes) Length of Frame. Excluding Framing Length Field.
+* __Frame Length__: (24 bits = max value 16,777,215) Positive integer representing the length of Frame in bytes. Excluding Framing Length Field.
```

Comment:
I disagree that it doesn't matter.

A signed or unsigned notion has nothing to do with programming languages. It just tells us how to interpret the bits.

So, here for example, if we say it is signed, and we get `Frame Length = 2^24 - 1`, we know something is wrong because we got a negative Frame Length. If we say it is unsigned, there are no wrong/error values.

My main point here is, that we need to separate between:
* how to interpret the bits - we do this by adding "signed" or "unsigned"
* any constraints on the value - e.g. "it MUST be positive"

In this specific instance, I suggest:

```diff
- Positive integer representing the length of Frame in bytes.
+ Unsigned integer representing the length of Frame in bytes.
```

In other instances, I suggest:

Stream ID: 
```diff
- Positive signed integer
+ Signed integer. MUST be non-negative.
```

Metadata Length:
```diff
- unsigned positive Integer
+ unsigned integer
```

Initial Request N:
> 32-bit signed integer representing the initial request N value for the stream. Only positive values are allowed.

This is actually a great example! The only thing I'd consider changing here is to use the [RFC 2119](https://tools.ietf.org/html/rfc2119) language (e.g. "MUST" instead of "allowed").